packages/core/src/Banner.js

```diff
@@ -56,11 +65,16 @@ const Banner = props => {
     bannerColors[
       !props.bg && props.color === 'green' ? props.color : props.bg
     ] || {}
-  const icon = props.iconName || bannerColor.icon
+  const Icon = props.icon || bannerColor.icon
   const color =
     hasPaletteColor(props) || !bannerColor.color
       ? props.color
       : bannerColor.color
+  const header = React.isValidElement(props.header) ? (
```

Comment:
I feel like this should first check if the `prop.header` exists, then check if it is a valid element or if we should create one, else header can be null.

Then we can remove the check on line 90 `{props.header && header}` and just use `{header}`, and then we aren't creating an empty header when `props.header` in undefined.